Purchases/FoundationExtensions/String+Extensions.swift

```diff
@@ -16,10 +16,29 @@ import Foundation
 
 extension String {
 
+    enum StringError: Error {
+
+        case escapingEmptyString
+
+    }
+
     func rot13() -> String {
         ROT13.string(self)
     }
 
+    func escapedOrError() throws -> String {
+        let trimmedAndEscapedAppUserID = self
```

Comment:
we won't only use this for app user ids, right? so maybe we can name this something more generic?